### PR TITLE
Introduce dependencies and rules for C and C++

### PR DESCRIPTION
**Pull request by krishnangovindraj**
## What is the goal of this PR?
Add google's GTEST as dependency for use in our C and C++ code in typedb-driver.

## What are the changes implemented in this PR?
* Introduces google's gtest library as a dependency.

### CODE REVIEW

File: library/cpp/clang_format.sh
Comment (dmitrii-ubskii) [inline library/cpp/clang_format.sh:1]: This should move to dependencies (or bazel-distribution) later

Comment (dmitrii-ubskii) [reply library/cpp/clang_format.sh:1]: Also, I think the way this is set up, this should run like maven update, i.e. format all the C files. In CI, we should have a job that runs this and fails on a non-empty git diff.

Comment (krishnangovindraj) [reply library/cpp/clang_format.sh:1]: (We discussed this in standup, but copying here)
This now runs more like a check style test, with the user running clang-format from their IDE.
Will add it to CI the same way we do checkstyle:
`bazel test $(bazel query 'kind(clang_format_test, //...)')`

Comment (krishnangovindraj) [reply library/cpp/clang_format.sh:1]: Why `bazel-distribution`
> Bazel rules for assembling and deploying software distributions 

Whereas the checkstyle rule & maven update rules both live here in dependencies.

**Review by dmitrii-ubskii — APPROVED** [review]

**Pull request merged**